diesel/src/upsert/mod.rs

```diff
@@ -4,6 +4,7 @@
 //!
 //! * PostgreSQL version 9.5 or newer
 //! * Sqlite3 version 3.24.0 or newer
+//! * MySQL version 8.0 or newer
```

Comment:
It appears as though this is supported on MySQL 5.7 as well! https://dev.mysql.com/doc/refman/5.7/en/insert-on-duplicate.html